data/tools/wesnoth_addon_manager

```diff
@@ -292,24 +302,34 @@ if __name__ == "__main__":
             args.upload = args.upload.rstrip("/")
 
             # New style with _server.pbl
-            pblfile = os.path.join(args.upload, "_server.pbl")
+            pbl_file_name = os.path.join(args.upload, "_server.pbl")
             name = os.path.basename(args.upload)
             wmldir = args.upload
             cfgfile = None # _main.cfg will be uploaded with the rest
             ignfile = os.path.join(args.upload, "_server.ign")
         else:
             # Old style with external .pbl file
-            pblfile = args.upload
+            pbl_file_name = args.upload
             name = os.path.basename(args.upload)
             name = os.path.splitext(name)[0]
             wmldir = os.path.join(os.path.dirname(args.upload), name)
             cfgfile = args.upload.replace(".pbl", ".cfg")
             ignfile = args.upload.replace(".pbl", ".ign")
 
         if args.pbl:
-            pblfile = args.pbl
+            pbl_file_name = args.pbl
+
+        with open(pbl_file_name, 'r') as pbl_file:
+            pbl_text = pbl_file.read()
+
+        for key_value_pair in args.pbl_key or []:
+            key = key_value_pair[0]
+            value = key_value_pair[1]
+            if not re.match("^[a-zA-Z]+$", key):
```

Comment:
This seems a little strict. Should at least allow underscores as well?

Reply:
TBH, _server.pbl doesn't really contain underscore, so it seems a bit excessive. After all, we don't support tags anyway (for example, `[feedback].topic_id`). But since it was written in review, fixed. Allowing underscores in the middle of the name now (not first symbol).